### ISSUE TITLE
GTFS-rt are empty quite often

### ISSUE THREAD

**Issue by gcamp** [SoundTransit/soundtransit-rds#25]
I have not seen a pattern of when/how this happens, but sometimes the data returned for the GTFS-rt is simply empty (only the header is present). 

Attached the .pb that I just retrieved from http://api.pugetsound.onebusaway.org/api/gtfs_realtime/trip-updates-for-agency/1.pb?key=my-key

https://www.dropbox.com/s/mx5wekvv2sgfyye/1.pb?dl=1

**Comment by gcamp** [comment]
Any news on this? It's a problem that's really important. Basically we can't use it.

**Comment by sheldonabrown** [comment]
Sorry, I thought this was closed out along with the issue about buses that
are on schedule are shown.

I'll add some monitoring to watch for it -- to see if I can narrow it down.

Sheldon

**Comment by gcamp** [comment]
Any news on this?

**Comment by sheldonabrown** [comment]
Yes. I've been running monitoring on it for a week now, and apart from maintenance windows everything has been fine. 

What feeds are you pulling from, and at what times are they empty?

**Comment by bdferris** [comment]
I've managed to repro this, but only once in 24 hours.  I've got a script pulling the KCM gtfs-realtime feed every 30 seconds.  I managed to get an empty feed at 1430289396 (2015-04-28 23:36:36 GMT-0700 (PDT)).

**Comment by sheldonabrown** [comment]
Thanks Brian!

Though I can't tell if that's a yay or a nay.  Once in a blue moon seems
like an acceptable loss to me :)

You wrote this, do you think there is a bug in the gtfs-rt exporter?
Anything I should look further into?

Sheldon

**Comment by bdferris** [comment]
Yeah, I think the NYC GTFS-realtime feed is much worse by comparison (they return a corrupted feed like 5% of the time).  Unless @gcamp is seeing this more regularly, I'm not sure how much resources to throw at this.

@sheldonabrown there could certainly be a bug.  I wrote this like 3 years ago hoping ST would turn it on a bit sooner ;)  I'm not sure what your architecture is these days, but I know the call to retrieve all active trips from the TransitDataService is not necessarily cheap.  If that's running via an RPC (from webapp frontend to TransitDataService on a backend service), then maybe there could be an occasional time-out?

Maybe the input GTFS-realtime feed from KCM is occasionally dropping out?  Not sure how OBA would react to that.

**Comment by sheldonabrown** [comment]
@bdferris A timeout from the TDS is possible though not something I've observed; hessian tends to get this right. (I don't believe the architecture has changed significantly since you've seen it last).

And the TDS internal state tends to survive an occasional KCM drop out, so that is less likely as well.

Looking through the logs for your example, the TDS does not report anything out of the ordinary, and there is nothing reported on the API servers, so no help there.


I have monitoring looking for an empty feed once a minute (previously I was only checking once every 5 minutes).  If the frequency of empty feeds increases then I'll escalate, but for now I'll simply watch it.

**Comment by bbodenmiller** [comment]
Are the GTFS-RT feeds dead or invalid again? I haven't been seeing any real-time data in Google Maps or Transit App for the last few days. In fact Transit is now showing real-time data for a trip from Monday evening rather than today.

**Comment by gcamp** [comment]
https://github.com/SoundTransit/soundtransit-rds/issues/48
